### DISCUSSION TITLE
How can I remap ACE editor shortcuts?

### DISCUSSION THREAD

**Discussion by mskar** [brookhong/Surfingkeys#1400] [Q&A] [answered]
I'd like to use `Ctrl-d` to delete right and `Ctrl-u` to delete to start of line in the ACE editor.
Currently, `Ctrl-d` deletes word left and `Ctrl-u` to uppercases word.
I tried the settings below without any luck.
```js
aceVimMap('<C-d>', '<Delete>', 'insert');
aceVimMap('<C-u>', '<C-o>d0', 'insert');
```
`Ctrl-d` still deletes word left and `Ctrl-u` simply stops working.
How can I get these bash/emacs style keybindings to work in the ACE editor?

**Comment by nemya9066** [comment, marked as answer]
I used this mapping to delete to start of the line:
aceVimMap('\<C-u>', '\<Esc>d0xi', 'insert');
You can use this to delete a character to the right but it won't work with \<C-d> which is what you want to use
aceVimMap('\<C-s>', '\<Esc>lcl', 'insert')
Apparently Ctrl keys that are already mapped can't be remapped If there is a function to unmap a key in Ace maybe it will solve the problem.
